IBAnimatable/AnimatableImageView.swift

```diff
@@ -110,16 +110,24 @@ import UIKit
   
   public override func layoutSubviews() {
     super.layoutSubviews()
-    
     autoRunAnimation()
   }
   
+  // MARK: - CALayer
+  public override class func layerClass() -> AnyClass {
+    return CAGradientLayer.self
+  }
+  
+  var gradientLayer: CAGradientLayer {
```

Comment:
I am not 100% on the idea to create a layer property for each `subclass`, can we encapsulate it in `DesignableGradientView`, `DesignableGradientView` is a subview of other UI elements conform to `GradientDesignable`.

There are some reason I can think of for encapsulating `gradientLayer` within `DesignableGradientView`
1. Better Autolayout support, CALayer doesn't support Autolayout directly, if we rotate the device, we may need to manipulate the layer again to fit the new orientation.
2. The subclasses e.g. `AnimatableImageView` do not need to have any knowledge of gradient implementation. According to [Open/closed principle](https://en.wikipedia.org/wiki/Open/closed_principle) 
3. Keep all interface consistent like `configXXXXX()` Although we have created a lot of UI elements start with `AnimatableXXXX`, but another goal of the project is to provide protocol extensions to create custom UI elements easily, e.g. `MaterialButton` or `BootstrapButton`. To achieve that we need to make all API as clean as possible. 

I feel the challenges when we work on a library, we need think about how to implement and at the same time, we also consider how the others use the library. But we get better architecture skills on the way while doing that. Let's have some fun with those challenges. Please let me know if you have any questions.